### DISCUSSION TITLE
iOS port / Common cpp core

### DISCUSSION THREAD

**Discussion by omkar-pai** [ThibaultBee/StreamPack#94] [Ideas]
First off, Thank you so much for your fantastic work on this library @ThibaultBee 

I had a few questions that i would love for some input on

- I was wondering if there are any plans for an iOS port of StreamPack / srtdroid. 
- I understand iOS development can be a pain sometimes, did that also play a part in keeping this library Android only? 
- Have you considered a common native core for Android and iOS? Please correct me if I am wrong but, I am guessing it would be easier to interface cpp to Objective C for iOS in comparison to cpp through JNI for Android.

 Thanks again.

**Comment by ThibaultBee** [comment]
Hi,

>     * I was wondering if there are any plans for an iOS port of StreamPack / srtdroid.

There isn't any plan to develop StreamPack or srtdroid on iOS.
What you need to know is that I am only working on StreamPack/srtdroid on my spare time and I don't have much these days. And I don't earn anything from this library.
Building this library for Android tooks a couple years. I don't have the time to do the same library on iOS. I would rather improve the Android library.

>     * I understand iOS development can be a pain sometimes, did that also play a part in keeping this library Android only?

Android video development is also a pain. Like when video encoders does not have the same behavior...
I did an Android library because I am an Android developer and I only had Android phones home.

>     * Have you considered a common native core for Android and iOS? Please correct me if I am wrong but, I am guessing it would be easier to interface cpp to Objective C for iOS in comparison to cpp through JNI for Android.

Right now, I don't have enough knowledge of C++. 
Howver I did a common lib in C in one the previous company I worked for.  Developing in C is so much slower than Kotlin or Swift that I am not sure anymore it will be faster. And I am not sure it was really performant. Well it needs lot of specific code to be performant.
But yeah, some parts like muxer could be in C or C++ but I like to know how they really work.

If I get time (which I won't), I would like to develop a KMM even if performance are not that great.

I hope it answers your questions. :)
